state/core_access.go

```diff
@@ -646,16 +635,12 @@ func (ca *CoreAccessor) getSigner(cfg *TxConfig) (AccAddress, error) {
 	}
 }
 
-// THIS IS A TEMPORARY SOLUTION!!!
-// unsetTx helps to fix issue in TxResponse marshaling. Marshaling TxReponse
-// fails because `TxResponse.Tx` is not empty but does not contain respective codec
-// for encoding using the standard `json.MarshalJSON()`. It becomes empty when
-// https://github.com/celestiaorg/celestia-core/issues/1281 will be merged.
-// The `TxResponse.Tx` contains the transaction that is sent to the cosmos-sdk in the form
-// in which it is processed there, so the user should not be aware of it.
-func unsetTx(txResponse *TxResponse) *TxResponse {
-	if txResponse != nil && txResponse.Tx != nil {
-		txResponse.Tx = nil
-	}
-	return txResponse
+// convertToTxResponse converts the user.TxResponse to sdk.TxResponse.
+// This is a temporary workaround in order to avoid breaking the api.
+func convertToSdkTxResponse(resp *user.TxResponse) *TxResponse {
+	return &TxResponse{
+		Code:   resp.Code,
+		TxHash: resp.TxHash,
+		Height: resp.Height,
+	}
```

Comment:
This seems to omit most of TxResponse fields. Is it guranteed that other fields are not used / send?

Reply:
where would the other fields be used? It omits most of TxResponse fields because we have a more lightweight tx indexer which only returns these fields hence making the process more efficient